game-app/game-core/src/main/java/games/strategy/triplea/attachments/RulesAttachment.java

```diff
@@ -507,6 +510,39 @@ private void resetIsAI() {
     isAI = null;
   }
 
+  @VisibleForTesting
+  public void setHasResource(final String value) throws GameParseException {
+    final String[] s = splitOnColon(value);
+    final int n = getInt(s[0]);
+    if (n < 1) {
+      throw new GameParseException("hasResource must be a positive number" + thisErrorMsg());
+    }
+    if (s.length == 1) {
+      hasResource = new String[] {value, "PUs"};
```

Comment:
Maybe the string `"PUs"` should be extracted into a constant? (e.g. `@NonNls public static final String DEFAULT_RESOURCE_UNIT`)